### PR TITLE
tpacpi-bat: init at 3.0

### PR DESCRIPTION
**Pull request by orbekk**
###### Motivation for this change

Allow setting battery thresholds on newer Thinkpad models that don't support the tp_smapi kernel module.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/maintainers.nix [hunk 1]
```diff
@@ -317,6 +317,7 @@
   olcai = "Erik Timan <dev@timan.info>";
   olejorgenb = "Ole Jørgen Brønner <olejorgenb@yahoo.no>";
   orbitz = "Malcolm Matalka <mmatalka@gmail.com>";
+  orbekk = "KJ Ørbekk <kjetil.orbekk@gmail.com>";
```
Comment (zimbatm) [inline lib/maintainers.nix]: orbekk goes before orbitz

**Review by zimbatm — APPROVED** [review]
LGTM overall

File: lib/maintainers.nix (hunk 1 above)
Comment (orbekk) [inline lib/maintainers.nix]: Done.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -11018,6 +11018,8 @@ in
 
     tp_smapi = callPackage ../os-specific/linux/tp_smapi { };
 
+    tpacpi-bat = callPackage ../os-specific/linux/tpacpi-bat { };
```
Comment (fpletz) [inline pkgs/top-level/all-packages.nix]: There is no need for this packages to be in `linuxPackages` as it's just a script that doesn't depend on any specific kernel for building. Only out-of-tree kernel modules are typically in `linuxPackages`.

File: pkgs/os-specific/linux/tpacpi-bat/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchgit, perl, kmod, acpi_call, coreutils }:
```
Comment (fpletz) [inline pkgs/os-specific/linux/tpacpi-bat/default.nix]: `acpi_call` & `coreutils` are not used below. Just drop them.

The `acpi_call`  module is a runtime dependency but that has to be specified in the NixOS configuration.

**Review by fpletz — CHANGES_REQUESTED** [review]
Works for me. Thanks! Please fix the minor issues and we'll merge. ;)

File: pkgs/os-specific/linux/tpacpi-bat/default.nix (hunk 3 above)
Comment (orbekk) [inline pkgs/os-specific/linux/tpacpi-bat/default.nix]: Done. Thanks for the review!

**Review by fpletz — APPROVED** [review]

**Comment by fpletz** [comment]
Pushed fixes some other things I found. You should use a version in the derivation name and fetchFromGitHub only fetches the tarball provided by Github instead of the git repository.

**Comment by fpletz** [comment]
Squashed our commits & merged. Also fixed the package name typo in the commit message.

**Comment by fpletz** [comment]
Thanks! :beers:

**Comment by makefu** [comment]
@orbekk is this the same `tpacpi-bat` which comes bundled with tlp ( /nix/store/*-tlp-0.8/lib/tlp-pm/tpacpi-bat )? If it is package is cooler than the tlp bundled tpacpi-bat then it might be a good idea to replace it when building `tlp` :)

**Comment by orbekk** [comment]
@makefu Thanks for pointing that out. I'll send a pull request to replace the one that comes bundled with TLP.

@fpletz fetchFromGithub is neat. I'll keep it in mind for the future.

**Comment by orbekk** [comment]
@makefu Just took another look at this. The tpacpi-bat that comes with tlp is only used internally, and I think replacing it is unnecessary and might break something.

**Comment by makefu** [comment]
@orbekk thanks for the heads-up :+1: let's just have two separate packages then

**Pull request merged**